main/2021/CG-03-30.md

```diff
@@ -42,4 +94,192 @@ Installation is required, see the calendar invite.
+
+SG: Procedurally, TC39 has a staging model, similar to the phases model for wasm. One of the requests from TC39 before stage 3 is to have integration with the Wasm CG worke dout. That we’re not gating stage 3 on the wasm side. Should the integration PR go through phases? Or once the integration is finalized, to have a nominal change?
+
+CW: motivation for Wasm side for going through phase structure is to make sure we have tests for JS API side before we move it forward.
+
+SG: before we move on to procedural question, any question about technical stuff?
+
+LW: sounds great. A resizable array buffer can’t be transferred via postmessage, can it?
+
+SG: They have the same detach key, so no. Only memory.grow can detach memory array buffers.
+
+DG: CW brought up a good point, want to make sure this has tests, aside of tests, can we do something more lightweight than phases model, we have no precedent, for the delta changes, mostly touching on JS API, we can do something like that
+
+DS: we can poll phase 1, and go directly to phase 3 once we have tests. And shoot for phase 4 around the time it goes through TC39.
+
+SG: sounds like a good suggestion, next TC39 is in less than a month, if we can get some kind of phase advancement, it will be a good signal for TC39, will ask for phase 3, then come back after some implementation work has been done, with tests, and final version of integration PR, further hash out details, something to play with behind a flag.
+
+DS: In the spec repo, we have all these core tests, do we also have JS tests? 
+
+ZN: <Yes, we do.>
+
+SG: What do we do when merging with the threads proposal? I don’t know quite how to write the PR for that. 
+
+DS: current thread proposal technically still at stage 3…
+
+CW: 2 in fact, we don’t have tests, we need more formal semantics to go to stage 4.
+
+DS: another option to merge this into threads proposal directly
+
+SG: the non-concurrent parts don’t need to be gated on threads proposal. For SharedArrayBuffer parts, the semantics on growing these growable shared array buffers basically designed to be the same across Wasm and JS. We’re both happy with results.
+
+CW: In terms of how we handle the spec, it may make sense to add the non-resizeable parts into the main spec, and then rebase the threads on that.
+
+DS: we do phase 1 today, write some tests, if they are ready in 2 weeks we can poll again to move it along, that should send a clear signal to TC39 that we can fast track the non-concurrent parts
+
+SG: sounds great to me
+
+
+
+
+Poll
+SF: 14
+F: 17
+N:
+A:
+SA:
+
+
+
+
+### Follow-on discussion on irreducible control flow [(previous slides)](https://docs.google.com/presentation/d/1KE7z7ry3VEug3nyIvnLwTtxxPhkp73CrYjpATpnNSF8/edit?usp=sharing) (Conrad Watt) [30 minutes]
+
+<CW Presenting [slides](https://docs.google.com/presentation/d/1KE7z7ry3VEug3nyIvnLwTtxxPhkp73CrYjpATpnNSF8/edit?usp=sharing)>
```

Comment:
```suggestion
<CW Presenting [slides](https://docs.google.com/presentation/d/1CloQonh4aVfCO-Z2yX98EyWFo3uiBwXkY0symhIwDIU/edit?usp=sharing)>
```

Reply:
link above was to previous week's slides, suggested change links to the slides I presented in the follow-on